core/templates/pages/exploration-editor-page/exploration-save-and-publish-buttons/exploration-save-and-publish-buttons.component.ts

```diff
@@ -129,13 +140,24 @@ angular.module('oppia').component('explorationSaveAndPublishButtons', {
         $scope.saveIsInProcess = false;
         $scope.publishIsInProcess = false;
         $scope.loadingDotsAreShown = false;
-
+        $scope.explorationCanBePublished = false;
         UserExplorationPermissionsService.getPermissionsAsync()
           .then(function(permissions) {
-            $scope.showPublishButton = function() {
-              return permissions.canPublish && $scope.isPrivate();
-            };
+            $scope.explorationCanBePublished = permissions.canPublish;
           });
+        ctrl.directiveSubscriptions.add(
+          ExplorationRightsService.onExplorationUnpublished.subscribe(
+            () => {
+              UserExplorationPermissionsService.getPermissionsAsync()
+                .then(function(permissions) {
+                  $scope.explorationCanBePublished = permissions.canPublish;
+                  // TODO(#8521): Remove the use of $rootScope.$apply()
+                  // once the controller is migrated to angular.
+                  $rootScope.$applyAsync();
+                });
```

Comment:
```              var updateExplorationPublishStatusAsync = () => { 
                UserExplorationPermissionsService.getPermissionsAsync()
                .then(function(permissions) {
                  $scope.explorationCanBePublished = permissions.canPublish;
                  // TODO(#8521): Remove the use of $rootScope.$apply()
                  // once the controller is migrated to angular.
                  $rootScope.$applyAsync();
                });
        }   
```

Use this function on lines 144 & 151.

Reply:
Sorry, I don't understand. Could you perhaps rephrase it?

Reply:
The GitHub comment wasn't working as expected earlier, I've updated the above comment, is that helpful?